src/box/lua/console.lua

```diff
@@ -768,7 +768,7 @@ local function start()
     end
     started = true
     local self = setmetatable({ running = true }, repl_mt)
-    local home_dir = os.getenv('HOME')
+    local home_dir = os.getenv_limited('HOME', os.PATH_MAX)
```

Comment:
Maybe, instead of introducing a new function, just add an optional argument to `os.getenv`? Also, we need to update our [documentation](https://www.tarantool.io/en/doc/latest/reference/reference_lua/osmodule/#os-getenv).

Reply:
> Maybe, instead of introducing a new function, just add an optional argument to os.getenv?

I don't know, give me some time to think. I feel like we shouldn't change existing functions signature or behaviour.

> Also, we need to update our [documentation](https://www.tarantool.io/en/doc/latest/reference/reference_lua/osmodule/#os-getenv).

Yep, thanks for the reminder. Added a docbot request.

Will add a test shortly as well.

Reply:
Here's the test:
```diff
diff --git a/test/app-luatest/getenv_safe_test.lua b/test/app-luatest/getenv_safe_test.lua
new file mode 100644
index 000000000..afd0b2bb2
--- /dev/null
+++ b/test/app-luatest/getenv_safe_test.lua
@@ -0,0 +1,15 @@
+local t = require('luatest')
+
+local g = t.group('getenv-safe')
+
+g.test_getenv_safe = function()
+    local env_name = 'GH_7977_GETENV_SAFE'
+    local env_value = 'some env string'
+    os.setenv(env_name, env_value)
+    t.assert_equals(os.getenv(env_name), env_value, "getenv sees the value")
+    t.assert_equals(os.getenv(env_name), os.getenv_safe(env_name, #env_value + 1),
+                    "getenv_safe returns the same result as getenv")
+    t.assert_equals(os.getenv_safe(env_name, #env_value), nil,
+                    "getenv_safe returns nil when result has too big size")
+    os.setenv(env_name, nil)
+end

```

Reply:
A new Lua function `os.getenv_safe` sounds good to me.